src/main/java/dude/Storage.java

```diff
@@ -0,0 +1,147 @@
+package dude;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+
+import dude.task.Deadline;
+import dude.task.Event;
+import dude.task.Task;
+import dude.task.Todo;
+
+/**
+ * Storage class that controls task storing functionalities of Dude.
+ */
+public class Storage {
+    private static final String directoryPath = "./data";
+    private String filePath = "./data/dude.txt";
+
+    /**
+     * Class constructor.
+     */
+    public Storage() {
+        boolean success = createStorageIfNotExists();
+        if (!success) {
+            throw new DudeNoStorageException();
+        }
+    }
+
+    /**
+     * Class constructor specifying the list name.
+     * @param listName List name which will be used as the path for the storage file.
+     */
+    public Storage(String listName) {
+        this.filePath = String.format("./data/%s.txt", listName);
+        boolean success = createStorageIfNotExists();
+        if (!success) {
+            throw new DudeNoStorageException();
+        }
+    }
+
+    private boolean createStorageIfNotExists() {
+        new File(Storage.directoryPath).mkdirs();
+        File storageFile = new File(this.filePath);
+        if (!storageFile.exists()) {
+            System.out.println("dude.Storage file not found. Creating storage file...");
```

Comment:
Would it be better to throw these lines up to the Ui to reflect to the user the state of the program too?